src/Server/Event/Listener/CollectionListener.php

```diff
@@ -0,0 +1,107 @@
+<?php
```

Comment:
Will you add a comment at the top here explaining what this listener is for and how it affects collections?